readme.md

```diff
@@ -10,248 +8,408 @@ See also: https://github.com/ga-wdi-lessons/ruby-oop
 - Define inheritance in the context of OOP
 - Write a Ruby class that inherits from another
 
-## `self` in ruby
+## Framing
 
-In ruby, self is very similar to `this` in Javascript, it refers to the object
-the method was called on. Thankfully, however, it's much simpler in Ruby...
-because methods can't be called without context (i.e. no callbacks), there's no
-`bind` to worry about, `self` always equals the thing the method was called on.
+[This morning](https://github.com/ga-wdi-lessons/ruby-oop) you started to learn about object-oriented programming in Ruby and how we can use classes to create a blueprint for objects containing related attributes and methods. Like constructor functions in Javascript, we can then use classes and the `.new` method to produce instances of these objects.
 
-### Exercise on `self` (15 minutes)
+This afternoon we're going to continue diving intro OOP in Ruby. In particular, we'll be taking a look at **inheritance** and how we can pass attributes and methods between classes.
 
-Pair up and consider the following example, specifically the `demonstrate_using_self`
-method. Talk with your partner for 7 minutes and discuss what effect each line
-will have, and which one is the best.
+But first, let's review and build upon some of the concepts that were introduced this morning...
 
-Feel free to run this code and play with it.
+## We Do: Let's Build A Class
 
-```ruby
-# person/self.rb
-class Person
-  attr_accessor :name
-  attr_reader :hunger_level
+Let's create a `Person` class...
 
-  def initialize(initial_name, initial_hunger_level)
-    @name = initial_name
-    @hunger_level = initial_hunger_level
+--------
+
+<details>
+  <summary><strong>What's the difference between a class and an instance?</strong></summary>
+  <br/>
+
+  A class is a blueprint. An instance is an object generated from that blueprint.
+  * A class' name is capitalized (e.g., `Person`) and is a template for objects of that class.
+  * An instance is one specific object created using a class (e.g. `bob = Person.new("Bob", 10)`).
+</details>
+<br/>
+<details>
+  <summary><strong>How do we begin defining a <code>Person</code> class?</strong></summary>
+
+  ```rb
+  class Person
   end
+  ```
+
+</details>
+
+--------
 
-  def hunger_level=(new_hunger_level)
-    if new_hunger_level < 0
-      @hunger_level = 0
-    else
-      @hunger_level = new_hunger_level
+<details>
+  <summary><strong>How can we set up our Person class so that it has <code>name</code> and <code>age</code> attributes. We should be able to read and modify <code>name</code>, but only read <code>age</code>.</strong></summary>
+
+  ```rb
+  class Person
+    # This allows us to both read and write to the name and hunger_level attributes
+    attr_accessor :name
+    attr_reader :age
```

Comment:
recommend introducing these separately.

e.g. start with:

1. getter and setter methods
2. attr_reader & writer
3. attr_accessor